xgitguard/file-scanner/secret_detection.py

```diff
@@ -0,0 +1,907 @@
+                secrets_data = secrets_data[:40]
+            secret_data_list = format_detection(
+                keyword,
+                file_path,
+                code_content,
+                secrets_data,
+                keyword_count,
+                is_secondary_credential,
+            )
+            if secret_data_list:
+                for secret_data in secret_data_list:
+                    secrets_data_list.append(secret_data)
+        else:
+            logger.debug(
+                f"Skipping secrets_data as length is not between 1 to 20. Length: {len(secrets_data)}"
+            )
+    else:
+        logger.debug(
+            f"Skiping processing code content is empty. Content length: {len(lines)}"
+        )
+
+    return secrets_data_list
+
+
+def run_detection(
+    file_path,
+    secondary_keyword,
+    secondary_credentials,
+    keyword,
+    extension,
+    ml_prediction=True,
+    code_content="",
+):
+    """
+    Run search with Secondary Keyword and extension combination in the given file
+    Steps:
+        Get the code content for the given file
+        Clean the code content and extract secrets
+        Detect the secrets using RegEx and format secret records
+        Predict the secret data using ML model
+        Write the cleaned and detected secret data
+
+    params: file_path - string
+    params: secondary_keyword - string
+    params: extension - string
+    params: ml_prediction - Boolean - optional - Default: True
+    params: code_content - string
+    params: secrets_data - list
+    returns: None
+    """
+    logger.debug("<<<< 'Current Executing Function' >>>>")
+    # Read and Setup Global Configuration Data to reference in all process
+    try:
+        global configs
+        if configs:
+            pass
+    except:
+        # Setting Global configuration Data
+        configs = ConfigsData()
+
+    global detection_writes_count
+
+    if not secondary_keyword and not secondary_credentials:
+        logger.error(f"Please pass secondary_keyword or credentials like 'password'")
+        sys.exit(1)
+
+    if not extension:
+        logger.error(f"Please pass extension like 'py'")
+        sys.exit(1)
+
+    if code_content:
+        logger.debug(
+            f"Processing Scan on secondary_keyword: '{secondary_keyword}' extension: '{extension}' file: '{file_path}'"
+        )
+        is_secondary_credential = None
+        if keyword in secondary_credentials:
+            is_secondary_credential = True
+        else:
+            is_secondary_credential = False
+
+        if is_secondary_credential:
+            code_contents = remove_url_from_creds(code_content, [])
+        else:
+            code_contents = remove_url_from_keys(code_content)
+
+        secrets_data = []
+        if code_contents:
+            if is_secondary_credential:
+                try:
+                    if configs.stop_words:
+                        pass
+                except:
```

Comment:
mention excpetion name as it handles all types of excpetions

Reply:
Added exception handling